### DISCUSSION TITLE
AWS Lambda - No package metadata was found for sqlalchemy

### DISCUSSION THREAD

**Discussion by T1Kap** [sqlalchemy/sqlalchemy#11420] [Usage Questions]
Hi,

I'm working on a python runtime upgrade from 3.8 to 3.10 on AWS Lambda.

This Lambda contains a layer deployed with Terraform with the following packages : 

- SQLAlchemy 1.3.24, 
- SQLAlchemy-Utils 0.34.1, 
- psycopg2

The lambda works fine with python3.8.
But since i changed runtime to python3.10, i had to upgrade SQLAlchemy-Utils to 0.38.3 for compatibility and then when i deploy my new lambda i have the following exception : 

"Exception: Exception invoking lambda : lbd-app-example{"errorMessage": "Unable to import module 'lambda_function': No package metadata was found for sqlalchemy", "errorType": "Runtime.ImportModuleError", "requestId": "5d2b9187-f76f-41e4-b6bf-72c79b47aze4", "stackTrace": []}"

I've tried so many things like upgrading or downgrading SQLAlchemy but nothing works for now.

I didn't change anything else except my docker python image to 3.10 that build the zip package to the lambda.

**Comment by CaselIT** [comment]
Hi,

Version 1.3 seems to have wheels only up to python 3.9, but you may be able to install it also on 3.10.
Try to very if locally you can install on python 3.10.
If not then your options would be:
- keep using 3.8 or try 3.9
- upgrade sqlalchemy

**Reply by T1Kap** [reply]
Hi,

The project works in local with docker, it seems it may be an issue related to AWS Lambda environnement..
I also tried to upgrade sqlAlchemy to the lasted 1.4 version but it doesn't change anythings.

**Reply by CaselIT** [reply]
Version 1.4 has wheels for python up to 3.12, so that should install without issues on 3.10.
If locally it works but on aws it does not, them you should likely look for support on the aws side
